### ISSUE TITLE
When predicting high-resolution photos, the bounding box cannot enclose the image.

### ISSUE THREAD

**Issue by NeKoooo233** [ultralytics/ultralytics#23240]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hello, When predicting high-resolution photos, the bounding box cannot enclose the image.
When I change the `imgsz` value in `test.py` from 640 to 1280 and 1600, the bounding box maintains a fixed size and gradually fails to encompass the object. When `imgsz=640`, all bounding box ranges are normal. Has this bug been fixed in later versions? I need to predict objects at least at 1280 resolution. 

### Environment

ultralytics=8.3.163

### Minimal Reproducible Example

test.py
from PIL import Image

from ultralytics import YOLO

# Load a pretrained YOLO26n model
model = YOLO("last.pt")

# Run inference on 'bus.jpg'
results = model.predict("
1.jpg", save=True, imgsz=1280, conf=0.25)

# Visualize the results
for i, r in enumerate(results):
    # Plot results image
    im_bgr = r.plot()  # BGR-order numpy array
    im_rgb = Image.fromarray(im_bgr[..., ::-1])  # RGB-order PIL image

    # Show results to screen (in supported environments)
    r.show()

    # Save results to disk
    r.save(filename=f"results{i}.jpg")

### Additional

test.py
from PIL import Image

from ultralytics import YOLO

# Load a pretrained YOLO26n model
model = YOLO("last.pt")

# Run inference on 'bus.jpg'
results = model.predict("
1.jpg", save=True, imgsz=1280, conf=0.25)

# Visualize the results
for i, r in enumerate(results):
    # Plot results image
    im_bgr = r.plot()  # BGR-order numpy array
    im_rgb = Image.fromarray(im_bgr[..., ::-1])  # RGB-order PIL image

    # Show results to screen (in supported environments)
    r.show()

    # Save results to disk
    r.save(filename=f"results{i}.jpg")

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NeKoooo233, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖🧩—an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You’ve included a script (thanks! 🙏). To make it a full MRE, please also include:
-Exact command you run (CLI or Python) and whether the issue occurs with `save=True`, `r.plot()`, or both
-The input image’s original resolution and a sample image (or a way to reproduce with a public image)
-A sample output image showing the incorrect boxes at `imgsz=1280/1600` vs correct at `imgsz=640`
-Model details: which YOLO model/task (`detect/segment/pose/obb`), and whether `last.pt` is custom-trained
-Any relevant logs printed during prediction

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can try YOLO26. It should be better with large objects.

**Comment by NeKoooo233** [comment]
> You can try YOLO26. It should be better with large objects.

Really? I want to refine the prediction of large objects. Can YOLOv26 draw normal bounding boxes? I remember discussing this with you yesterday; I thought it was a problem with the degree parameter. Has the bounding box drawing bug been fixed in YOLOv26?

**Comment by Y-T-G** [comment]
YOLO26 doesn't DFL which is the major reason for this

**Comment by NeKoooo233** [comment]
> YOLO26 doesn't DFL which is the major reason for this

Okay, I'll try it. Thank you.

**Comment by NeKoooo233** [comment]
> YOLO26 doesn't DFL which is the major reason for this

YOLOv26's prediction performance on keypoints is simply amazing. Using the RLE loss function, it truly learns the features of the keypoints, rather than simply regressing their positions.

**Comment by glenn-jocher** [comment]
Good to hear YOLO26 pose is behaving well there (credit to the broader YOLO community and the Ultralytics team); if you still see any bounding-box scaling/plotting issues at `imgsz=1280+`, please update to the latest `ultralytics` and open a new issue with the full text output of `yolo checks` plus a minimal repro/sample image so we can reproduce—refs: [Ultralytics YOLO26 docs](https://docs.ultralytics.com/models/yolo26/) and [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/).
